From e9a4a7f3bd4d5cf3bee15ea1cdbd9d9fc9ff0f5b Mon Sep 17 00:00:00 2001
From: Mikhail Paulyshka <me@mixaill.net>
Date: Tue, 18 Aug 2026 17:03:45 +0300
Subject: [PATCH 1/1] gr-qtgui: actually run tests through xvfb-run

Restore the intended behavior from pull request #3545. The
QA_DISPLAY_SERVER variable is set when xvfb-run is found, but accidentally was dropped from the gr_add_test invocation, causing qa_qtgui to run without a display.

Signed-off-by: Mikhail Paulyshka <me@mixaill.net>
---
 gr-qtgui/python/qtgui/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gr-qtgui/python/qtgui/CMakeLists.txt b/gr-qtgui/python/qtgui/CMakeLists.txt
index 2d4c7f12e..177135a7e 100644
--- a/gr-qtgui/python/qtgui/CMakeLists.txt
+++ b/gr-qtgui/python/qtgui/CMakeLists.txt
@@ -61,7 +61,7 @@ if(ENABLE_TESTING)
         if(CMAKE_CROSSCOMPILING)
             get_filename_component(py_qa_test_file ${py_qa_test_file} NAME)
         endif(CMAKE_CROSSCOMPILING)
-        gr_add_test(${py_qa_test_name} ${QA_PYTHON_EXECUTABLE} -B ${py_qa_test_file})
+        gr_add_test(${py_qa_test_name} ${QA_DISPLAY_SERVER} ${QA_PYTHON_EXECUTABLE} -B ${py_qa_test_file})
     endforeach(py_qa_test_file)
 endif(ENABLE_TESTING)
 
-- 
2.55.0

